includes/core-changes/aspnetcore/5.0/blazor-components-trim-insignificant-whitespace.md

```diff
@@ -0,0 +1,73 @@
+### Blazor: Insignificant whitespace trimmed from components at compile time
+
+Starting with ASP.NET Core 5.0 Preview 7, the Razor compiler omits insignificant whitespace in Blazor components (*.razor* files) at compile time. For discussion, see issue [dotnet/aspnetcore#23568](https://github.com/dotnet/aspnetcore/issues/23568).
+
+#### Version introduced
+
+5.0 Preview 7
+
+#### Old behavior
+
+In 3.x versions of Blazor Server and Blazor WebAssembly, *.razor* files preserved whitespace in the text context of the rendered output. Consider the following Blazor component code:
+
+```razor
+<ul>
+    @foreach (var item in Items)
+    {
+        <li>
+            @item.Text
+        </li>
+    }
+</ul>
+```
+
+The preceding example renders two whitespace nodes:
+
+* Outside of the `@foreach` code block.
+* Around the `<li>` element.
+* Around the `@item.Text` output.
+
+A list containing 100 items results in 402 whitespace nodes. That's over half of all nodes rendered, even though none of the whitespace nodes visually affect the rendered output.
+
+When rendering static HTML for components, whitespace inside a tag header wasn't preserved. For example, view the source of the following rendered component. You'll notice that whitespace wasn't preserved.
```

Comment:
It looks to me like whitespace **is** preserved

Reply:
@tdykstra If you put `<foo        bar="baz"     />` inside a `.razor` file, then the prerendered output should look something like `<foo bar="baz" />`. Are you finding otherwise?

Reply:
I'm not using Blazor at all, I'm just judging from what I see in the text and example.  The text says there will be no whitespace, and then the example has whitespace.

Reply:
I guess the original phrasing is confusing. It says `view the source of the following rendered component`, and then gives source code for a component (*not* the rendered output). The source code itself isn't rendered; you're meant to render that source code and then see there's no whitespace in the result.

Over to @scottaddie for possibly clarifying the original phrasing :)